fluentui_notification/src/main/java/com/microsoft/fluentui/tokenized/notification/Snackbar.kt

```diff
@@ -49,21 +71,44 @@ class SnackbarMetadata(
     val subTitle: String?,
     val actionText: String?,
     val duration: NotificationDuration,
-    private val continuation: CancellableContinuation<NotificationResult>
+    private val continuation: CancellableContinuation<NotificationResult>,
+    val animationBehavior: AnimationBehavior
 ) : NotificationMetadata {
 
     override fun clicked() {
         try {
             if (continuation.isActive) continuation.resume(NotificationResult.CLICKED)
-        } catch (e: Exception){
+        } catch (e: Exception) {
+            // This can happen if there is a race condition b/w two events. In that case, we ignore the second event.
+        }
+    }
+
+    override fun clicked(scope: CoroutineScope) {
+        try {
+            scope.launch {
+                animationBehavior.onClickAnimation()
+                if (continuation.isActive) continuation.resume(NotificationResult.CLICKED)
+            }
+        } catch (e: Exception) {
             // This can happen if there is a race condition b/w two events. In that case, we ignore the second event.
         }
     }
 
     override fun dismiss() {
         try {
             if (continuation.isActive) continuation.resume(NotificationResult.DISMISSED)
-        } catch (e: Exception){
+        } catch (e: Exception) {
+            // This can happen if there is a race condition b/w two events. In that case, we ignore the second event.
+        }
+    }
+
+    override fun dismiss(scope: CoroutineScope) {
```

Comment:
can we do like this so we don't require to deperate pervious experince 

``` 
override fun dismiss(scope: CoroutineScope? = null) {
    try {
        if (scope != null) {
            scope.launch {
                animationBehavior.onDismissAnimation()
                if (continuation.isActive) continuation.resume(NotificationResult.DISMISSED)
            }
        } else {
            if (continuation.isActive) continuation.resume(NotificationResult.DISMISSED)
        }
    } catch (e: Exception) {
        // This can happen if there is a race condition b/w two events. In that case, we ignore the second event.
    }
}
```

Reply:
makes sense! Thanks for the suggestion! Changes done, resolving the comment.